### PR TITLE
Review 153

### PR DESCRIPTION
**Pull request by atomizable**

### CODE REVIEW

File: Reviews/153 - Uniswap Clone for Casper/1 - Contract Port 4 Engineers + 2 QA Engineers/README.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+We are able to deploy the uniswap contract and get it working with a compatible erc20.
+
+## Milestone Submission
+
+The following milestone assets/artifacts were submitted for review:
+
+Repository | Revision Reviewed
+------------ | -------------
+https://github.com/Rengo-Labs/uniswap-v2-router-casper | YES
```
Comment (mssteuer) [inline Reviews/153 - Uniswap Clone for Casper/1 - Contract Port 4 Engineers + 2 QA Engineers/README.md:32]: This should have the git revision, not "yes"

File: Reviews/153 - Uniswap Clone for Casper/1 - Contract Port 4 Engineers + 2 QA Engineers/README.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+They have a table of contents with links and type descriptions. 
+They far are ahead of their milestones in this capacity.
+Excellent. 
+
+### Code Documentation
+
+Next to the fantastic supporting documentation the inline documentation is exceedingly adaquate.
+But again, for this stage of the process, this is more than sufficient.
+The reviewer does urges the OP to include much more inline documentation in the source code. 
+This would be very helpful for untrained newcomers whom actually want to read the soruce.
```
Comment (mssteuer) [inline Reviews/153 - Uniswap Clone for Casper/1 - Contract Port 4 Engineers + 2 QA Engineers/README.md:76]: "Source"

File: Reviews/153 - Uniswap Clone for Casper/1 - Contract Port 4 Engineers + 2 QA Engineers/README.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+# Coding Standards
+
+Acceptable. 
+
+## General Observations
+
+# Final Conclusion
+
+It has not been mentioned yet, but two other reviews have been done: 
+ { Quantstamp's 1st Report, https://github.com/atomizable/dxd_codereview/blob/main/153%20-%20Uniswap%20Clone%20for%20Casper/Uniswap%20-%20Report.pdf }
```
Comment (mssteuer) [inline Reviews/153 - Uniswap Clone for Casper/1 - Contract Port 4 Engineers + 2 QA Engineers/README.md:139]: You should use a relative link to the artifact, as this link will stop working once you close the review and delete the branch

File: Reviews/153 - Uniswap Clone for Casper/1 - Contract Port 4 Engineers + 2 QA Engineers/README.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+Quantstamp will be issueing another report shortly. 
+This should be addressed in future reviews by future reviewers. 
+
+The reviewer has confirmed that the uniswap contract can be deployed within the same account. It shoud be noted that there 
+is a limitation with the CLI with contracts deployed different to accounts. 
+This seems to be due to a issue in the deployent namespace as contracts cannot easily be told thier own names. 
+As this is an ecosystem problem that the team understands well,
+it is the opinion of thie reviewer that the acceptance cirteria for a deploying a uniswap contract has been met. 
+It has been demonstracted to the reviewer that they have it working with two erc20s.
```
Comment (mssteuer) [inline Reviews/153 - Uniswap Clone for Casper/1 - Contract Port 4 Engineers + 2 QA Engineers/README.md:157]: Bunch of typos and misspelling

**Review by mssteuer — APPROVED** [review]
Adding feedback for correction but not withholding approval assuming you address these.

File: Reviews/153 - Uniswap Clone for Casper/1 - Contract Port 4 Engineers + 2 QA Engineers/README.md [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+
+## Overall Conclusion on Documentation
+
+An impressive display for this stage. 
+
+# Open Source Practices
+
+## Licenses
+
+The Project is released under the Apache License 2.0
```
Comment (mrkara) [inline Reviews/153 - Uniswap Clone for Casper/1 - Contract Port 4 Engineers + 2 QA Engineers/README.md:103]: GPL-3.0 License

File: Reviews/153 - Uniswap Clone for Casper/1 - Contract Port 4 Engineers + 2 QA Engineers/README.md [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+The following milestone assets/artifacts were submitted for review:
+
+Repository | Revision Reviewed
+------------ | -------------
+https://github.com/Rengo-Labs/uniswap-v2-router-casper | 18f1576 
+https://github.com/Rengo-Labs/uniswap-v2-core-casper | 5b5f0b2
+
+# Install & Usage Testing Procedure and Findings
+
+The furnished instructions we more than adaquate for this stage in the project and for the target audience(advanced users).
```
Comment (mrkara) [inline Reviews/153 - Uniswap Clone for Casper/1 - Contract Port 4 Engineers + 2 QA Engineers/README.md:37]: were

File: Reviews/153 - Uniswap Clone for Casper/1 - Contract Port 4 Engineers + 2 QA Engineers/README.md [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+## Contribution Policies
+
+The current state of the code is in private repositories. 
+As this code is sensitive in nature at the moment, it is the opinion of this reviewer that this is 
+appropriate, given the circumstances. If a malicious actor was to gain access to this code and set 
+up a DEX, the reputation of the Casper Ecosystem could be affected. 
+
+The OP has indicated that he is willing to show the 
+repository to any individual that may ask. 
+In light og the high-profile nature of the project,
```
Comment (mrkara) [inline Reviews/153 - Uniswap Clone for Casper/1 - Contract Port 4 Engineers + 2 QA Engineers/README.md:119]: light of

File: Reviews/153 - Uniswap Clone for Casper/1 - Contract Port 4 Engineers + 2 QA Engineers/README.md [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+
+The current state of the code is in private repositories. 
+As this code is sensitive in nature at the moment, it is the opinion of this reviewer that this is 
+appropriate, given the circumstances. If a malicious actor was to gain access to this code and set 
+up a DEX, the reputation of the Casper Ecosystem could be affected. 
+
+The OP has indicated that he is willing to show the 
+repository to any individual that may ask. 
+In light og the high-profile nature of the project, 
+this seems reasonable and nessecary and this reveiwer
```
Comment (mrkara) [inline Reviews/153 - Uniswap Clone for Casper/1 - Contract Port 4 Engineers + 2 QA Engineers/README.md:120]: necessary

File: Reviews/153 - Uniswap Clone for Casper/1 - Contract Port 4 Engineers + 2 QA Engineers/README.md [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+
+## Overall Conclusion on Documentation
+
+An impressive display for this stage. 
+
+# Open Source Practices
+
+## Licenses
+
+The Project is should be released under the Apache License 2.0 License as set forth in the
```
Comment (mrkara) [inline Reviews/153 - Uniswap Clone for Casper/1 - Contract Port 4 Engineers + 2 QA Engineers/README.md:103]: The Project should be

**Review by mrkara — APPROVED** [review]
LGTM

**Pull request merged**